compositor_pipeline/src/pipeline/decoder/fdk_aac.rs

```diff
@@ -0,0 +1,202 @@
+use compositor_render::{AudioSamples, AudioSamplesBatch, InputId};
+use crossbeam_channel::{Receiver, Sender};
+use fdk_aac_sys as fdk;
+
+use crate::pipeline::structs::{EncodedChunk, EncodedChunkKind};
+
+use super::AacDecoderOptions;
+
+struct Decoder(*mut fdk::AAC_DECODER_INSTANCE);
+
+#[derive(Debug, thiserror::Error)]
+pub enum AacDecoderError {
+    #[error("The internal fdk decoder returned an error: {0:?}.")]
+    FdkDecoderError(fdk::AAC_DECODER_ERROR),
+
+    #[error("The channel config in the aac audio is unsupported.")]
+    UnsupportedChannelConfig,
+
+    #[error("The aac decoder cannot decode chunks with kind {0:?}.")]
+    UnsupportedChunkKind(EncodedChunkKind),
+}
+
+impl Decoder {
+    fn new(options: AacDecoderOptions) -> Result<Self, AacDecoderError> {
+        // MP4_RAW should cover a huge majority of MP4 AAC streams (95+%), and detecting which transport
+        // is used in the mp4 seems to be very hard
```

Comment:
It actually isn't that hard, the information is just so scattered in various specifications that it was very hard to find. I know how to do this now and will implement it in a  subsequent PR.